ably/realtime_conn.go

```diff
@@ -588,11 +597,16 @@ func (c *Connection) advanceSerial() {
 func (c *Connection) send(msg *protocolMessage, onAck func(err error)) {
 	hasMsgSerial := msg.Action == actionMessage || msg.Action == actionPresence
 	c.mtx.Lock()
+	// RTP16a - in case of presence msg send, check for connection status and send accordingly
 	switch state := c.state; state {
 	default:
 		c.mtx.Unlock()
 		if onAck != nil {
-			onAck(connStateError(state, nil))
+			if c.state == ConnectionStateClosed {
+				onAck(errClosed)
+			} else {
+				onAck(connStateError(state, nil))
```

Comment:
@SimonWoolf re-raising as per comments mentioned here https://github.com/ably/ably-go/pull/621/files#r1412500532

Reply:
resolving the conversation, since we don't want to emit error when connection is closed as such